### PR TITLE
add Max Temperature Feature

### LINKED ISSUE #34: Add feature for finding max temperature

### PR DESCRIPTION
**Pull request by SergiLomtadze**
closes #34
Unit and Integration Tests will be in next PR

### CODE REVIEW

File: src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/MaxWeather/MaxWeatherCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        private List<string> GetCitiesFromInput()
+        {
+            var cities = new List<string>();
+            var inputedLine = _rwOperation.ReadLine();
+            string[] words = inputedLine.Split(',');
+            foreach (string word in words)
+            {
+                cities.Add(word.Trim());
+            }
+            return cities;
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/MaxWeather/MaxWeatherCommandHandler.cs:71]: Then, this method simplifies to
```
return _rwOperation.ReadLine().Split(',')
```
which makes this method unnecessary. Consider inlining it

File: src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/WeatherApiService.cs [hunk 2]
```diff
@@ -104,5 +105,45 @@ public async Task<List<City>> GetFutureTemperatureByCoordinateAndDayQuantity(Coo
...
+                return new MaxTempCityInfo
+                {
+                    Name = name,
+                    ErrorMessage = $"City: {name} was not found"
+                };
+                //throw new NotFoundException($"City: {name} was not found");
+            }
+            else if (result.IsSuccessStatusCode)
+            {
+                var json = result.Content.ReadAsStringAsync().Result;
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/WeatherApiService.cs:127]: OK... try to avoid using Task.Result at all.

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]

File: src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/MaxWeather/MaxWeatherCommandHandler.cs (hunk 1 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/MaxWeather/MaxWeatherCommandHandler.cs:71]: but, between commas may also be spaces, and I need to parse(trim) each of  them after splitting, that's why I can't make it inline.

File: src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/MaxWeather/MaxWeatherCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                }
+                else
+                {
+                    _rwOperation.WriteLine($"City: {maxTempCityInfo.Name}. Error: {maxTempCityInfo.ErrorMessage}");
+                }
+            }
+                
+        }
+
+        private List<string> GetCitiesFromInput()
```
[the remark this answers is not in the document]
Comment (SergiLomtadze) [reply src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/MaxWeather/MaxWeatherCommandHandler.cs:62]: done

File: src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/MaxWeather/MaxWeatherCommandHandler.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            foreach (var city in cities)
+            {
+                tasks.Add(_weatherApiService.GetTemperatureByCityNameForMaxTemp(city));
+            }
+
+            await Task.WhenAll(tasks);
+            var maxTempCityInfoList = new List<MaxTempCityInfo>();
+            foreach (var task in tasks)
+            {
+                maxTempCityInfoList.Add(task.Result);
```
[the remark this answers is not in the document]
Comment (SergiLomtadze) [reply src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/MaxWeather/MaxWeatherCommandHandler.cs:40]: done

File: src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/WeatherApiService.cs (hunk 2 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/WeatherApiService.cs]: done

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]

**Review by kdavletov-exadel — APPROVED** [review]

**Pull request merged**